src/components/Warnings/modals/WarningTrackerModal.jsx

```diff
@@ -0,0 +1,398 @@
+            </p>
+            <p className="modal__information">
+              Pressing the "Add New Warning Tracker" button will allow you to add a new warning to
+              the list.
+            </p>
+          </Popover.Content>
+        </Popover>
+      );
+    }
+    return (
+      <Popover id="details">
+        <Popover.Title as="h4">Description</Popover.Title>
+        <Popover.Content>
+          This will {title} this warning tracker,{' '}
+          {title === 'activate'
+            ? 'showing all saved warning-tracking data of this type'
+            : 'retaining the data but hiding all warning tracking of this type'}
+        </Popover.Content>
+      </Popover>
+    );
+  };
+
+  const handleTriggerDeleteWarningDescription = warning => {
+    setToggleDeleteModal(true);
+    setWarningToDelete(warning);
+  };
+
+  const handleDeactivate = warningId => {
+    dispatch(updateWarningDescription(warningId)).then(res => {
+      if (res.error) {
+        setError(res.error);
+        return;
+      }
+      setWarningDescriptions(prev =>
+        prev.map(warning => {
+          if (warning._id === warningId) {
+            return { ...warning, activeWarning: !warning.activeWarning };
+          }
+          return warning;
+        }),
+      );
+      getUsersWarnings();
+    });
+  };
+  const handleDeleteWarningDescription = warningId => {
+    dispatch(deleteWarningDescription(warningId)).then(res => {
+      if (res.error) {
+        setError(res.error);
+        return;
+      }
+      setWarningDescriptions(prev => prev.filter(warning => warning._id !== warningId));
+      getUsersWarnings();
+    });
+  };
+
+  const handleEditWarningDescription = (e, warningId) => {
+    setWarningEdited(true);
+
+    const updatedWarningDescriptions = warningDescriptions.map(warning => {
+      if (warning._id === warningId) {
+        const updatedWarning = { ...warning, warningTitle: e.target.value };
+        setEditedWarning(updatedWarning);
+        return updatedWarning;
+      }
+      return { ...warning, disabled: true };
+    });
+    setWarningDescriptions(updatedWarningDescriptions);
+  };
+
+  const handleCancelEdit = () => {
+    fetchWarningDescriptions();
+    setWarningEdited(false);
+    setWarningWasEdited(false);
+  };
+
+  const handleSaveEditedWarning = () => {
+    dispatch(editWarningDescription(editedWarning)).then(res => {
+      if (res.error) {
+        setError(res.error);
+        setWarningEdited(false);
+        setEditedWarning(null);
+        fetchWarningDescriptions();
+        return;
+      }
+      setWarningEdited(false);
+      setEditedWarning(null);
+      setError(null);
+      getUsersWarnings();
+      fetchWarningDescriptions();
+      setWarningWasEdited(true);
+    });
+  };
+
+  // eslint-disable-next-line no-shadow
+  const handleAddNewWarning = (e, newWarning) => {
+    e.preventDefault();
+
+    if (newWarning === '') return;
+    const trimmedWarning = newWarning.trim();
```

Comment:
I would suggest adding frontend validation for special characters, even though backend already had it.